### ISSUE TITLE
NewlineAtEndOfFile: update doc to mention that it allows any number of empty lines at the end

### ISSUE THREAD

**Issue by posuhov** [checkstyle/checkstyle#18159]
I have read check documentation: https://checkstyle.org/checks/misc/newlineatendoffile.html#NewlineAtEndOfFile
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac TestNewLine.java
Compilation successful

/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>

<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Package Names Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="NewlineAtEndOfFile"/>
</module>

/var/tmp $ cat TestNewLine.java
public class TestNewLine {

}






/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-12.1.2-all.jar -c config.xml TestNewLine.java
Starting audit...
Audit done.
```

The issue with NewlineAtEndOfFile rule is that it checks for **at least 1 empty line** at the end of file. That means if we have 10 empty lines at the end of class we do not get an error. Either documentation should be updated or rule should be fixed

**Comment by romani** [comment]
>it checks for at least 1 empty line at the end of file

I don't see this in our documentation.

Do you mean that we should extend our doc to me mention that Check doesn't do violation on few consecutive new lines at the end of file?

**Comment by posuhov** [comment]
> > it checks for at least 1 empty line at the end of file
> 
> I don't see this in our documentation.
> 
> Do you mean that we should extend our doc to me mention that Check doesn't do violation on few consecutive new lines at the end of file?

hi @romani, thank you for quick answer. I see several options here:
1) if it's expected that this check for **at least 1 emty line** at the end, then this should be mentioned in the rule documentation
2) if this is not expected behavior, then rule should be fixed to check for **exactly 1 empty line** at the enf of the file.
3) the rule could be extended with  parameters `min-lines` defaulting to 1 and `max-lines` defaulting to null(which means any). 

I supposed that it works as described in 2nd option, but was surprised when found it doesn't fail on 2 empty lines as well. However 3 option looks most flexible and backward compartible with current approach

**Comment by romani** [comment]
Please send PR to extend documentation. We should fix it by option 1. 
Validation of duplicate empty lines, should be done by some other check, I think we already shared somewhere in issue tracking a way to do this. We try to keep Check to do single task, I can not approve extension of Check.

Place to update:
https://github.com/checkstyle/checkstyle/blob/343cc19f1fb8e7dc0695481cc6a8751e3eac93ce/src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java#L75

**Comment by romani** [comment]
@posuhov , please pin me here if you can not find how to put violations on duplicate empty lines, I will try to find.

**Comment by ekansh0unofficial** [comment]
I can take this up. Preparing a PR.

**Comment by romani** [comment]
Doc is updated at https://github.com/checkstyle/checkstyle/pull/18203
